command/var_get.go

```diff
@@ -0,0 +1,182 @@
+
+  If ACLs are enabled, this command requires a token with the 'var:read'
+  capability.
+
+General Options:
+
+  ` + generalOptionsUsage(usageOptsDefault) + `
+
+Read Options:
+
+  -item <item key>
+     Print only the value of the given item. Specifying this option will
+     take precedence over other formatting directives. The result will not
+     have a trailing newline making it ideal for piping to other processes.
+
+  -out ( go-template | hcl | json | none | pretty-json | table )
+     Format to render the variable in. When using "go-template", you must
+     provide the template content with the "-template" option. Defaults
+     to "table" when stdout is a terminal and to "json" when stdout is
+     redirected.
+
+  -template
+     Template to render output with. Required when output is "go-template".
+
+`
+	return strings.TrimSpace(helpText)
+}
+
+func (c *VarGetCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(c.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-out":      complete.PredictSet("table", "hcl", "json", "pretty-json", "go-template"),
+			"-template": complete.PredictAnything,
+		},
+	)
+}
+
+func (c *VarGetCommand) AutocompleteArgs() complete.Predictor {
+	return VariablePathPredictor(c.Meta.Client)
+}
+
+func (c *VarGetCommand) Synopsis() string {
+	return "Read a variable"
+}
+
+func (c *VarGetCommand) Name() string { return "var read" }
+
+func (c *VarGetCommand) Run(args []string) int {
+	var out, item string
+
+	flags := c.Meta.FlagSet(c.Name(), FlagSetClient)
+	flags.Usage = func() { c.Ui.Output(c.Help()) }
+	if fileInfo, _ := os.Stdout.Stat(); (fileInfo.Mode() & os.ModeCharDevice) != 0 {
+		flags.StringVar(&c.outFmt, "out", "table", "")
+	} else {
+		flags.StringVar(&c.outFmt, "out", "json", "")
+	}
+	flags.StringVar(&c.tmpl, "template", "", "")
+	flags.StringVar(&item, "item", "", "")
+
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	// Check that we got one or two arguments
+	args = flags.Args()
+	if len(args) != 1 {
+		c.Ui.Error("This command takes one argument: <path>")
+		c.Ui.Error(commandErrorText(c))
+		return 1
+	}
+
+	if err := c.validateOutputFlag(); err != nil {
+		c.Ui.Error(err.Error())
+		c.Ui.Error(commandErrorText(c))
+		return 1
+	}
+
+	path := args[0]
+
+	// Get the HTTP client
+	client, err := c.Meta.Client()
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+		return 1
+	}
+
+	qo := &api.QueryOptions{
+		Namespace: c.Meta.namespace,
+	}
+
+	sv, _, err := client.Variables().Read(path, qo)
+	if err != nil {
+		if err.Error() == "variable not found" {
+			c.Ui.Warn(errVariableNotFound)
+			return 1
+		}
+		c.Ui.Error(fmt.Sprintf("Error retrieving variable: %s", err))
+		return 1
+	}
+	// If the user provided an item key, return that value instead of the whole
+	// object
+	if item != "" {
+		if v, ok := sv.Items[item]; ok {
+			fmt.Print(v)
```

Comment:
Should this be `c.Ui.Output`?

Reply:
c.Ui.Output was forcing a terminal linefeed on me, where in this case I want the output to be as raw as possible.

Reply:
Ah! Right, makes sense.